pkg/machinery/scaffold.go

```diff
@@ -41,32 +49,96 @@ var options = imports.Options{
 }
 
 // Scaffold uses templates to scaffold new files
-type Scaffold interface {
```

Comment:
Why was this changed to a struct?

Reply:
Because we were exporting a `Option` that was a function on an unexported struct.

Reply:
You can do `type ScaffoldOption func(*scaffold)` since the predefined options have access to the internal type.

Reply:
I would prefer ^ but it's an unimportant design decision.

Reply:
That means that we are sporting `ScaffoldOption` which has a unexported parameter. I avoided this when able.

Reply:
That's a common practice. The set of options is fixed because `Scaffold`'s fields are unexported anyway.

Reply:
Oh, yes, I know. The result is exactly the same. But there was not added value by the interface either. And this way we don't have a unexported type in pkg.go.dev.